File: lists/lesson3.py
# create a list and loop over the list
looping_list = ["Shepard", 6, True]
for item in looping_list:
    print(item)

# create a list and add one to each item of the list. Does not mutate the item.
non_mutate = [1, 2, 3, 4]
new_non = non_mutate.copy()
for x in non_mutate:
    new_non.append(x + 1)
print(new_non)


# create a list and add one to each item of the list. Make sure it mutates the original list.
mutate = [10, 20, 30, 40, 50]
updated = [num+1 for num in mutate]  #THANK YOU PYTHON DOCUMENTATION FOR LIST COMPREHENSION !!
print(updated)

#original block I wrote for line 14. end result wasn't what I think was wanted
# mutate = [10, 20, 30, 40, 50]
# for num in mutate:
#     updated = []
#     num += 1
#     updated.append(num)
#     print(updated)

# create a list and loop backwards through the list
backwards = [10, 9, 8, 7, 6, 5, 4, 3, 2, 1]
for i in reversed(backwards):
    print(i)

# create a list and only print out every other item
even = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
every_other = even[1::2]   #[start:end:step] 
print(every_other)

File: arrays/lesson3.js
// # create a array and loop over the array
const array1 = [1, 2, 3, 4, 5];
for (let num of array1) {
    console.log(num);
}

// # create a array and add one to each item of the array. Does not mutate the item.
const array2 = [1, 2, 3, 4, 5];
const array3 = []
for (let num of array2) {
    num += 1;
    array3.push(num);
}
console.log(array2);
console.log(array3);

// # create a array and add one to each item of the array. Make sure it mutates the original array.
const array4 = [10, 20, 30, 40];
for (num = 0; num < array4.length; num++) {
    array4[num] += 1;
}
console.log(array4);

// # create a array and loop backwards through the array
const array5 = ["backwards", "is", "array", "This"];
for (let item of array5.reverse()) {
    console.log(item);
}

// # create a array and only print out every other item
const array6 = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
for (num = 0; num < array6.length; num+=2) {
    console.log(array6[num]);
}